### PR TITLE
Add skyloong/Dt40 keyboard

### PR DESCRIPTION
**Pull request by JZ-Skyloong**
## Description
Add skyloong/Dt40 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/skyloong/dt40/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        KC_ESC,     KC_Q,    KC_W,     KC_E,    KC_R,   KC_T,   KC_Y,     KC_U,      KC_O,      KC_I,    KC_BSPC,
+       KC_CAPS,     KC_A,    KC_S,     KC_D,    KC_F,   KC_G,   KC_H,     KC_J,      KC_L,      KC_P,     KC_ENT,
+       KC_LSFT,     KC_Z,    KC_X,     KC_C,    KC_V,   KC_B,   KC_N,     KC_M,   KC_RSFT,     KC_UP,    RGB_TOG,
+       KC_LCTL,  KC_LALT,  KC_SPC,  KC_LWIN,  KC_SPC,   KC_K, KC_SPC, 	KC_APP,   KC_LEFT,   KC_DOWN,    KC_RGHT
+    ),
+
```
Comment (waffle87) [inline keyboards/skyloong/dt40/keymaps/default/keymap.c:34]:
```suggestion
const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
    /* Base */
    [0] = LAYOUT(
        KC_ESC,     KC_Q,    KC_W,     KC_E,    KC_R,   KC_T,   KC_Y,     KC_U,      KC_O,      KC_I,    KC_BSPC,
       KC_CAPS,     KC_A,    KC_S,     KC_D,    KC_F,   KC_G,   KC_H,     KC_J,      KC_L,      KC_P,     KC_ENT,
       KC_LSFT,     KC_Z,    KC_X,     KC_C,    KC_V,   KC_B,   KC_N,     KC_M,   KC_RSFT,     KC_UP,    RGB_TOG,
       KC_LCTL,  KC_LALT,  KC_SPC,  KC_LWIN,  KC_SPC,   KC_K, KC_SPC, 	KC_APP,   KC_LEFT,   KC_DOWN,    KC_RGHT
    )
```

File: keyboards/skyloong/dt40/readme.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make skyloong/dt40:default
+    make skyloong/dt40:via
+
+Flashing example for this keyboard:
+
+    make skyloong/dt40:default:flash
+    make skyloong/dt40:via:flash
```
Comment (fauxpark) [inline keyboards/skyloong/dt40/readme.md:24]:
```suggestion

Flashing example for this keyboard:

    make skyloong/dt40:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by JZ-Skyloong — COMMENTED** [review]
Files were modified regarding to suggesting.

**Comment by JZ-Skyloong** [comment]
QMK CI check shows a lack of two files crt1.td and chmtx.td, how can I correct it?

**Comment by JZ-Skyloong** [comment]
PR has not been updated for a long time, what can I do next? @fauxpark @waffle87  @lesshonor  @drashna

**Review by waffle87 — APPROVED** [review]

**Pull request merged**